### ISSUE TITLE
BTSubTree prints an error for an unset tree before _enter

### ISSUE THREAD

**Issue by onze** [limbonaut/limboai#94]
My tree contains a BTSubTree that I want to load dynamically. In order to do this, I leave the subtree field unset, and a sibling custom node does this:
```
var target_subtree: BTSubtree = find_btask(
	bt_player.get_tree_instance(), 
	my_subtree_task_name
) as BTSubtree
# assert target_subtree != null
target_subtree.subtree = null
var new_btree := ... # roughly load('res://...') as BehaviorTree
target_subtree.subtree = new_btree
```

In the editor, _before_ this runs, I get the following error:
```
my_file.gd:95 @ <anonymous lambda>(): Subtree root task is not valid.
  <C++ Error>    Condition "!subtree->get_root_task().is_valid()" is true.
  <C++ Source>   limboai/bt/tasks/decorators/bt_subtree.cpp:33 @ initialize()
```

And when my code above runs, the task doesn't update properly.

**Workaround**: I created a dummy btree and assigned it to the BTSubTree, so that the error isn't raised and the subtree can be replaced. It all works, I just wanted to raise a use case that wasn't taken into account. I think 2 minor changes could make this more user-friendly (I'm not aware of possible side-effects though):

- the error could be raised in `_enter`, rather than in initialization step
- setting `subtree` to a valid value should cancel the failed state of the task.

**Comment by limbonaut** [comment]
There was a discussion some time ago: #58 
It is currently not supported, as `BTSubtree` is designed to be a simple loader. However, you can do it dynamically, you just need to do it without `BTSubtree`, either by using a custom decorator, or by walking the tree and inserting your branch where you need it to be. Here's a simple example decorator (written from memory):
```gdscript
extends BTDecorator
## MyCustomBranch

func _setup() -> void:
    var bt: BehaviorTree = load("res://...")
    var my_branch: BTTask = bt.instantiate(agent, blackboard)
    add_child(my_branch)

func _tick(delta) -> Status:
    var child: BTTask = get_child(0)
    return child.execute(delta)
```

You may or may not want to isolate the blackboard scope for your subtree like this:
```gdscript
var new_scope := Blackboard.new()
new_scope.set_parent(blackboard)
var my_branch: BTTask = bt.instantiate(agent, new_scope)
```

**Comment by onze** [comment]
Makes sense, thanks for the quick answer. I opened [a PR](https://github.com/limbonaut/limboai/pull/96) to add a quick mention to this in BTSubtree's doc itself. Feel free to dismiss it if you feel like it's too much detail.

**Comment by speakk** [comment]
The example provided here seems to be quite outdated is bt.instantiate has a different signature now, and also gives an error such as: `Cannot assign a value of type BTInstance to variable "my_branch" with specified type BTTask.`

I'll see if I can get this working myself (very new with limboAI but need this feature), but otherwise any insight from the creator would be appreciated.

**Comment by limbonaut** [comment]
@speakk Check out the following link. It's an up-to-date example of how to create and manage a BT instance. 
https://github.com/limbonaut/limboai-extra/blob/main/examples/custom_bt_player/custom_bt_player.gd

You can also find information about methods in the documentation.

**Comment by limbonaut** [comment]
@speakk You probably don't need to create `BTInstance` for this use-case. Try `bt.get_root_task().clone()` instead. That should fix that example, I think.

**Comment by speakk** [comment]
Thank you I will try this out!
